presto-common/src/main/java/com/facebook/presto/common/block/AbstractMapBlock.java

```diff
@@ -190,8 +192,8 @@ public long getRegionLogicalSizeInBytes(int position, int length)
 
         return getRawKeyBlock().getRegionLogicalSizeInBytes(entriesStart, entryCount) +
                 getRawValueBlock().getRegionLogicalSizeInBytes(entriesStart, entryCount) +
-                (Integer.BYTES + Byte.BYTES) * length +
-                Integer.BYTES * HASH_MULTIPLIER * entryCount;
+                (Integer.BYTES + Byte.BYTES) * (long) length +
+                Integer.BYTES * HASH_MULTIPLIER * (long) entryCount;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).